### PR TITLE
`azurerm_windows_web_app`, `azurerm_windows_web_app_slot` - add `handler_mapping` property.

### PR DESCRIPTION
**Pull request by xiaxyi**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

--- PASS: TestAccWindowsWebApp_handlerMapping (218.50s)
--- PASS: TestAccWindowsWebApp_handlerMappingUpdate (556.04s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    584.993s

The failure is not relate to the code change in this pr
=== CONT  TestAccWindowsWebAppSlot_handlerMappingsUpdate
    testcase.go:113: Step 1/8 error: After applying this test step, the plan was not empty.
        stdout:

        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # azurerm_windows_web_app_slot.test will be updated in-place
          ~ resource "azurerm_windows_web_app_slot" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/85b3dbca-5974-4067-9669-67a141095a76/resourceGroups/acctestRG-240416171042070886/providers/Microsoft.Web/sites/acctestWA-240416171042070886/slots/acctestWAS-240416171042070886"
                name                                           = "acctestWAS-240416171042070886"
              ~ webdeploy_publish_basic_authentication_enabled = false -> true
                # (16 unchanged attributes hidden)

                # (1 unchanged block hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccWindowsWebAppSlot_handlerMappingsUpdate (257.20s)
--- PASS: TestAccWindowsWebAppSlot_handlerMappings (317.09s)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_windows_web_app`, `azurerm_windows_web_app_slot` - add `handler_mapping` property.

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @xiaxyi. I left some comments in-line, we can take another look through once they've been resolved and get this merged.

File: internal/services/appservice/helpers/common_web_app_schema.go [hunk 1]
```diff
@@ -933,6 +987,29 @@ func ExpandConnectionStrings(connectionStringsConfig []ConnectionString) *webapp
 	return result
 }
 
+func expandHandlerMapping(handlerMapping []HandlerMappings) *[]webapps.HandlerMapping {
+	result := make([]webapps.HandlerMapping, 0)
+	if len(handlerMapping) == 0 {
+		return &result
```
[the remark this answers is not in the document]
Comment (xiaxyi) [reply internal/services/appservice/helpers/common_web_app_schema.go:993]: Thanks @stephybun for the suggestion, we can't send a nil back as nil won't override the current value if the property is assigned with an actual value. The empty list will reset the property back to nil state.

[the remark this answers is not in the document]
Comment (stephybun) [reply internal/services/appservice/helpers/common_web_app_schema.go:993]: This scenario only appears to apply when an update is done? In which case we need a separate expand function for the update as I previously mentioned and like we do for other blocks e.g. https://github.com/hashicorp/terraform-provider-azurerm/blob/143dd9820bac06857f4d66b379bcbdfaa8182c48/internal/services/appservice/helpers/common_web_app_schema.go#L965

**Comment by xiaxyi** [comment]
Thanks @stephybun for the review, code is updated:
1. changed the property name from `script_processor` to `script_processor_path`
2. updated the expandFunction for handlerMapping, a dedicated expandForUpdate is added -> will follow the same practice in th e future.
3. updated docs: corrected grammer, adding block to data source doc.
4. acc test result, failure is not related:
```
=== CONT  TestAccWindowsWebApp_handlerMappingUpdate
    testcase.go:113: Step 1/8 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # azurerm_windows_web_app.test will be updated in-place
          ~ resource "azurerm_windows_web_app" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/xxx/resourceGroups/acctestRG-240418155706396311/providers/Microsoft.Web/sites/acctestWA-240418155706396311"
                name                                           = "acctestWA-240418155706396311"
              ~ webdeploy_publish_basic_authentication_enabled = false -> true
                # (18 unchanged attributes hidden)

                # (1 unchanged block hidden)
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
=== NAME  TestAccWindowsWebApp_handlerMapping
    testcase.go:113: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # azurerm_windows_web_app.test will be updated in-place
          ~ resource "azurerm_windows_web_app" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/xxx/resourceGroups/acctestRG-240418155706393666/providers/Microsoft.Web/sites/acctestWA-240418155706393666"
                name                                           = "acctestWA-240418155706393666"
--- FAIL: TestAccWindowsWebApp_handlerMappingUpdate (183.76s)
--- FAIL: TestAccWindowsWebApp_handlerMapping (186.52s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    206.834s
FAIL
make: *** [GNUmakefile:99: acctests] Error 1
=== CONT  TestAccWindowsWebAppSlot_handlerMappingsUpdate
=== NAME  TestAccWindowsWebAppSlot_handlerMappings
    testcase.go:113: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # azurerm_windows_web_app.test will be updated in-place
          ~ resource "azurerm_windows_web_app" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/xxx6/resourceGroups/acctestRG-240418155707110505/providers/Microsoft.Web/sites/acctestWA-240418155707110505"
                name                                           = "acctestWA-240418155707110505"
              ~ webdeploy_publish_basic_authentication_enabled = false -> true
                # (18 unchanged attributes hidden)

                # (1 unchanged block hidden)
            }

          # azurerm_windows_web_app_slot.test will be updated in-place
          ~ resource "azurerm_windows_web_app_slot" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/xxx/resourceGroups/acctestRG-240418155707110505/providers/Microsoft.Web/sites/acctestWA-240418155707110505/slots/acctestWAS-240418155707110505"
                name                                           = "acctestWAS-240418155707110505"
              ~ webdeploy_publish_basic_authentication_enabled = false -> true
                # (16 unchanged attributes hidden)

                # (1 unchanged block hidden)
            }

        Plan: 0 to add, 2 to change, 0 to destroy.
=== NAME  TestAccWindowsWebAppSlot_handlerMappingsUpdate
    testcase.go:113: Step 1/8 error: After applying this test step, the plan was not empty.
        stdout:


        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # azurerm_windows_web_app.test will be updated in-place
          ~ resource "azurerm_windows_web_app" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/xxx/resourceGroups/acctestRG-240418155707112866/providers/Microsoft.Web/sites/acctestWA-240418155707112866"
                name                                           = "acctestWA-240418155707112866"
              ~ webdeploy_publish_basic_authentication_enabled = false -> true
                # (18 unchanged attributes hidden)

                # (1 unchanged block hidden)
            }

          # azurerm_windows_web_app_slot.test will be updated in-place
          ~ resource "azurerm_windows_web_app_slot" "test" {
              ~ ftp_publish_basic_authentication_enabled       = false -> true
                id                                             = "/subscriptions/xxx/resourceGroups/acctestRG-240418155707112866/providers/Microsoft.Web/sites/acctestWA-240418155707112866/slots/acctestWAS-240418155707112866"
                name                                           = "acctestWAS-240418155707112866"
              ~ webdeploy_publish_basic_authentication_enabled = false -> true
                # (16 unchanged attributes hidden)
--- FAIL: TestAccWindowsWebAppSlot_handlerMappingsUpdate (244.29s)
--- FAIL: TestAccWindowsWebAppSlot_handlerMappings (251.15s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    272.192s
FAIL
make: *** [GNUmakefile:99: acctests] Error 1
```

Let me know if everything is ok to you.

**Review by stephybun — COMMENTED** [review]
Thanks @xiaxyi! Everything looks good, except it currently isn't possible to verify that updating a web app or web app slot with a `handler_mapping` actually works as expected since the test returns a non empty plan and can't proceed with the other test steps. Could you either
1. Update the test configs to prevent the diff 
or 
2. Look into why we're getting a plan diff?

Once we're able to run the tests and provided they pass it should be good to go then.

**Comment by xiaxyi** [comment]
Thanks @stephybun , the tests are passed after adding the block

```
lifecycle {
    ignore_changes = [
      webdeploy_publish_basic_authentication_enabled, ftp_publish_basic_authentication_enabled
    ]
  }
```

```
--- PASS: TestAccWindowsWebApp_handlerMapping (204.48s)
--- PASS: TestAccWindowsWebApp_handlerMappingUpdate (520.16s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    549.969s
--- PASS: TestAccWindowsWebAppSlot_handlerMappingsUpdate (784.68s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    788.730s
--- PASS: TestAccWindowsWebAppSlot_handlerMappings (286.66s)


```

**Review by stephybun — APPROVED** [review]
Thanks @xiaxyi LGTM 👍

**Pull request merged**